scripts/Install-VSFeatures.ps1

```diff
@@ -0,0 +1,108 @@
+param (
+	[Parameter(Mandatory=$true)]
+	[string[]] $Components,
+
+	[uri] $InstallerUri = "https://download.visualstudio.microsoft.com/download/pr/c4fef23e-cc45-4836-9544-70e213134bc8/1ee5717e9a1e05015756dff77eb27d554a79a6db91f2716d836df368381af9a1/vs_Enterprise.exe",
+
+	[string] $VsInstaller = "${env:System_DefaultWorkingDirectory}\vs_Enterprise.exe",
+
+	[string] $VsInstallOutputDir = "${env:System_DefaultWorkingDirectory}\vs",
+
+	[System.IO.FileInfo] $VsInstallPath = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\2019\Enterprise",
+
+	[System.IO.FileInfo] $VsInstallerPath = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer",
+
+	[switch] $Collect = $false,
+
+	[switch] $Cleanup = $false,
+
+	[switch] $UseWebInstaller = $false
+)
+
+$Components | ForEach-Object {
+	$componentList += '--add', $_
+}
+
+$LocalVsInstaller = "$VsInstallerPath\vs_installershell.exe"
+
+$UseWebInstaller = $UseWebInstaller -or -not (Test-Path -Path "$LocalVsInstaller")
+
+if ($UseWebInstaller) {
+	Write-Host "Downloading web installer..."
+
+	Invoke-WebRequest -Method Get `
+		-Uri $InstallerUri `
+		-OutFile $VsInstaller
+
+	New-Item -ItemType directory -Path $VsInstallOutputDir
+
+	Write-Host "Running web installer to download requested components..."
+
+	Start-Process `
+		-FilePath "$VsInstaller" `
+		-ArgumentList ( `
+			'--layout', "$VsInstallOutputDir",
+			'--wait',
+			'--norestart',
+			'--quiet' + `
+			$componentList
+		) `
+		-Wait `
+		-PassThru
```

Comment:
>-PassThru [](start = 2, length = 9)

Nit: Doesn't this end up writing to stdout? If so, it seems a bit at odds with the Write-Host calls you use elsewhere in the script (afaik, they go straight to screen and are not redirectable).